straxen/corrections_services.py

```diff
@@ -156,17 +155,25 @@ def get_pmt_gains(self, run_id, model_type, global_version,
         to_pe = None
         cache_name = None
 
-        if model_type == 'to_pe_model':
+        if 'to_pe_model' in model_type:
+            # Get the detector name based on the requested model_type
+            # This also will be used to the cachable name convention
+            # pmt == TPC, n_veto == n_veto's PMT, etc
+            detector_names = {'to_pe_model': 'pmt',
+                              'to_pe_model_nv': 'n_veto',
+                              'to_pe_model_mv': 'mu_veto'}
+            target_detector = detector_names[model_type]
+
             if global_version in cacheable_versions:
                 # Try to load from cache, if it does not exist it will be created below
-                cache_name = cacheable_naming(run_id, model_type, global_version)
+                cache_name = cacheable_naming(run_id, model_type, global_version, target_detector)
```

Comment:
Technically, you don't really need the `target_detector` as this is already in the `model_type`. I would remove it (target_detector).

Reply:
yes, you are right